libs/libc/netdb/Kconfig

```diff
@@ -237,6 +237,11 @@ config NETDB_DNSSERVER_IPv4ADDR
 		Default DNS server IPv4 address in host byte order.  Default value
 		10.0.0.1.  This may be changed via dns_add_nameserver().
 
+config NETDB_DNS_STREAM
```

Comment:
the original code already support both TCP/UDP, why do you make this patch?

Reply:
the simplest solution, if it's acceptable for your configuration, would be to bump your CONFIG_NETDB_DNSCLIENT_MAXRESPONSE.

i agree it makes sense to try to use malloc when it doesn't fit RECV_BUFFER_SIZE.
it shouldn't be too difficult to make our client decide after receiving the record size (the first two bytes of the response) with tcp.

i'm not sure how difficult buffer recycling is.

depending on your configurations and environment, you might also want to implement EDNS buffer size.
some servers "tweak" their responses to fit the specified size.

Reply:
thank you for your suggestion！